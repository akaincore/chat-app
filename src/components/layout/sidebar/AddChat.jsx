import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';

const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
    position: 'absolute',
    bottom: '55px',
    right: '10px',
  },
});

const AddChat = props => {
  const { classes } = props;
  return (
    <Button variant="fab" color="primary" aria-label="Add" className={classes.button}>
      <AddIcon/>
    </Button>
  );
};

export default withStyles(styles)(AddChat);
