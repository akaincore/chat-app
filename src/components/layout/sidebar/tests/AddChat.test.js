/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AddChat from '../AddChat';

jest.mock('../../../chat/AddChatModal', () => () => 'AddChatModal');

const mockProps = {
  handleModalClose: jest.fn(),
  onTitleChange: jest.fn(),
  handleCreateChat: jest.fn(),
  handleModalOpen: jest.fn(),
  modalOpen: false,
  disabled: false,
  title: 'foo',
};

describe('<AddChat />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddChat {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AddChat {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
