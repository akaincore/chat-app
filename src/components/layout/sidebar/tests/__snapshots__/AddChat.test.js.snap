// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddChat /> renders correctly 1`] = `
<div>
  <button
    aria-label="Add"
    className="MuiButtonBase-root-42 MuiFab-root-32 MuiFab-primary-34 AddChat-button-31"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiFab-label-33"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-45"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-54"
    />
  </button>
  AddChatModal
</div>
`;
